src/main/java/duke/data/Duke.java

```diff
@@ -0,0 +1,129 @@
+package src.main.java.duke.data;
+
+
+import src.main.java.duke.commands.Command;
+import src.main.java.duke.commands.CommandResult;
+import src.main.java.duke.data.task.Task;
+import src.main.java.duke.data.task.TaskList;
+import src.main.java.duke.data.task.TaskList.TaskNotFoundException;
+import src.main.java.duke.parser.Parser;
+import src.main.java.duke.storage.StorageFile;
+import src.main.java.duke.storage.StorageFile.StorageOperationException;
+
+/**
+ * Duke class represents the bot which contains a tasklist and interacts with the tasks.
+ */
+public class Duke {
+
+    private final TaskList taskList;
+
+    private StorageFile storageFile;
+
+    /**
+     * Creates an empty task list.
+     */
+    public Duke() {
+        taskList = new TaskList();
+    }
+    /**
+     * Constructs an task list with the given data.
+     *
+     * @param tasks external changes to this will not affect this task list
+     */
+    public Duke(TaskList tasks) {
+        this.taskList = new TaskList(tasks);
+    }
+
+    /**
+     * Sets the storage file for duke.
+     * @param storageFile storageFile which is the own storage file
+     * @return return the duke object
+     */
```

Comment:
JDocs guidelines suggest a space to be an empty line between the description and parameter section.